server/core/slug.service.ts

```diff
@@ -0,0 +1,26 @@
+import ValidationException from "./exceptions/validation.exception";
+import { replaceAll } from "./util";
+
+const forbiddenCharacters = new RegExp(/[^a-zA-Z0-9 ]/g)
+const duplicateWhitespace = new RegExp(/\s{2,}/g)
+
+export function createSlug(name: string) {
+    const trimmed = name.toLowerCase().trim()
+
+    if (trimmed.length < 1) {
+        throw new ValidationException("Slug can't be contructed from this name")
+    }
+
+    const withoutForbiddenCharacters = trimmed
+        .replace(forbiddenCharacters, "")
+
+    const withNormalizedWhitespace = withoutForbiddenCharacters
+        .replace(duplicateWhitespace, " ")
+
+    const slug = replaceAll(withNormalizedWhitespace, " ", "-")
+        .trim()     
+
+    return slug
```

Comment:
This now means we can't have two events with the same name?
Should we be supporting two 'Barcelona - PSG' matches at different timestamps?
If so, maybe we should consider concatenating an uuid or some other unique random identifier to non-unique-constrained slugs, e.g. https://stackoverflow.com/a/40191779
BTW, if we decide to tackle this issue, we can merge this and just put that in the new ticket.

Reply:
Yep, and there's even a database constraint to enforce this. But I didn't think it was a problem, as only us admins would be able to add events, so we can always change a name (prepend it with a year or something) to make a different slug if needed.

Random UUIDs would work (or just plain number IDs), but I wouldn't like to add them to all URLs, just the conflicting ones. 

But for now I've just added a new field in the event form, where you can override what the generated slug would be.

Reply:
Awesome :+1: